hudi-client/src/main/java/org/apache/hudi/client/HoodieWriteClient.java

```diff
@@ -586,24 +602,39 @@ public String startCommit() {
    * @param instantTime Instant time to be generated
    */
   public void startCommitWithTime(String instantTime) {
+    HoodieTableMetaClient metaClient = createMetaClient(true);
+    startCommitWithTime(instantTime, metaClient.getCommitActionType(), metaClient);
+  }
+
+  /**
+   * Completes a new commit time for a write operation (insert/update/delete) with specified action.
+   */
+  public void startCommitWithTime(String instantTime, String actionType) {
+    HoodieTableMetaClient metaClient = createMetaClient(true);
+    startCommitWithTime(instantTime, actionType, metaClient);
+  }
+
+  /**
+   * Completes a new commit time for a write operation (insert/update/delete) with specified action.
+   */
+  private void startCommitWithTime(String instantTime, String actionType, HoodieTableMetaClient metaClient) {
     // NOTE : Need to ensure that rollback is done before a new commit is started
     if (rollbackPending) {
       // Only rollback inflight commit/delta-commits. Do not touch compaction commits
       rollbackPendingCommits();
     }
-    startCommit(instantTime);
+    startCommit(instantTime, actionType, metaClient);
   }
 
-  private void startCommit(String instantTime) {
-    LOG.info("Generate a new instant time " + instantTime);
-    HoodieTableMetaClient metaClient = createMetaClient(true);
+  private void startCommit(String instantTime, String actionType, HoodieTableMetaClient metaClient) {
```

Comment:
Does this have to be non-static ? Can it be moved to CommitUtils ?

Reply:
This is a private method. Do you want to make this public static?  Personally, I think having all startCommit methods in HoodieWriteClient makes more sense because user workflow is

1) writeClient#startCommit
2) writeClient#upsert
3) writeClient#commit

But if you have a strong preference to make this part of CommitUtils, I can move it. let me know.

Reply:
Lets leave it for now If needed, we can refactor later.